frame/dapp-staking-v3/src/lib.rs

```diff
@@ -0,0 +1,450 @@
+        /// Total locked amount for staker is below minimum threshold.
+        LockedAmountBelowThreshold,
+        /// Cannot add additional locked balance chunks due to size limit.
+        TooManyLockedBalanceChunks,
+    }
+
+    /// General information about dApp staking protocol state.
+    #[pallet::storage]
+    pub type ActiveProtocolState<T: Config> =
+        StorageValue<_, ProtocolState<BlockNumberFor<T>>, ValueQuery>;
+
+    /// Counter for unique dApp identifiers.
+    #[pallet::storage]
+    pub type NextDAppId<T: Config> = StorageValue<_, DAppId, ValueQuery>;
+
+    /// Map of all dApps integrated into dApp staking protocol.
+    #[pallet::storage]
+    pub type IntegratedDApps<T: Config> = CountedStorageMap<
+        _,
+        Blake2_128Concat,
+        T::SmartContract,
+        DAppInfo<T::AccountId>,
+        OptionQuery,
+    >;
+
+    /// General locked/staked information for each account.
+    #[pallet::storage]
+    pub type Ledger<T: Config> =
+        StorageMap<_, Blake2_128Concat, T::AccountId, AccountLedgerFor<T>, ValueQuery>;
+
+    /// General information about the current era.
+    #[pallet::storage]
+    pub type CurrentEraInfo<T: Config> = StorageValue<_, EraInfo<BalanceOf<T>>, ValueQuery>;
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// Used to register a new contract for dApp staking.
+        ///
+        /// If successful, smart contract will be assigned a simple, unique numerical identifier.
+        #[pallet::call_index(0)]
+        #[pallet::weight(Weight::zero())]
+        pub fn register(
+            origin: OriginFor<T>,
+            owner: T::AccountId,
+            smart_contract: T::SmartContract,
+        ) -> DispatchResult {
+            Self::ensure_pallet_enabled()?;
+            T::ManagerOrigin::ensure_origin(origin)?;
+
+            ensure!(
+                !IntegratedDApps::<T>::contains_key(&smart_contract),
+                Error::<T>::ContractAlreadyExists,
+            );
+
+            ensure!(
+                IntegratedDApps::<T>::count() < T::MaxNumberOfContracts::get().into(),
+                Error::<T>::ExcededMaxNumberOfContracts
+            );
+
+            let dapp_id = NextDAppId::<T>::get();
+            // MAX value must never be assigned as a dApp Id since it serves as a sentinel value.
+            ensure!(dapp_id < DAppId::MAX, Error::<T>::NewDAppIdUnavailable);
+
+            IntegratedDApps::<T>::insert(
+                &smart_contract,
+                DAppInfo {
+                    owner: owner.clone(),
+                    id: dapp_id,
+                    state: DAppState::Registered,
+                    reward_destination: None,
+                },
+            );
+
+            NextDAppId::<T>::put(dapp_id.saturating_add(1));
+
+            Self::deposit_event(Event::<T>::DAppRegistered {
+                owner,
+                smart_contract,
+                dapp_id,
+            });
+
+            Ok(().into())
```

Comment:
```suggestion
            Ok(())
```

`into` isn't needed anymore with `DispatchResult` as return type. Same to other calls.

Reply:
Right, I rushed! Will clean it up!